File: src/main/java/dev/scratch/scheduler/model/actions/HardAction.java
package dev.scratch.scheduler.model.actions;

import dev.scratch.scheduler.util.TimeFrame;

import java.time.DayOfWeek;

public class HardAction extends Action {
    private TimeFrame timeFrame;

    public HardAction(String content, TimeFrame timeFrame, DayOfWeek[] days) {
        super(content, days);
        this.timeFrame = timeFrame;
    }

    public HardAction(String content, TimeFrame timeFrame, DayOfWeek[] days, int intervalBefore, int intervalAfter) {
        super(content, days, intervalBefore, intervalAfter);
        this.timeFrame = timeFrame;
    }

    public TimeFrame getTimeFrame() {
        return timeFrame;
    }

    public HardAction setTimeFrame(TimeFrame timeFrame) {
        this.timeFrame = timeFrame;
        return this;
    }


}

File: src/main/java/dev/scratch/scheduler/model/actions/HardAction.kt
package dev.scratch.scheduler.model.actions

import dev.scratch.scheduler.util.TimeFrame
import java.time.DayOfWeek

open class HardAction(
    content: String, val timeFrame: TimeFrame, days: Array<DayOfWeek>, intervalBefore: Int = 15, intervalAfter: Int = 15
) : Action(content, days, intervalBefore, intervalAfter) {
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as HardAction

        if (timeFrame != other.timeFrame || content != other.content)
            return false

        return true
    }

    override fun hashCode(): Int {
        return timeFrame.hashCode() + content.hashCode()
    }


}